testutil/fstest/compare.go

```diff
@@ -35,3 +38,17 @@ func CheckDirectoryEqual(d1, d2 string) error {
 
 	return nil
 }
+
+// CheckDirectoryEqualWithApplier compares directory against applier
+func CheckDirectoryEqualWithApplier(a Applier, d string) error {
+	// TODO: cache applied
```

Comment:
Could you explain this more, not sure what the use case for caching the temp directory is.